httpclient/httpclient.go

```diff
@@ -0,0 +1,69 @@
+package httpclient
+
+import (
+	"crypto/tls"
+	"crypto/x509"
+	"fmt"
+	"github.com/gokrazy/internal/config"
+	"io/ioutil"
+	"log"
+	"net/http"
+	"net/url"
+	"os"
+	"path"
+	"strings"
+)
+
+func GetTLSHttpClientByTLSFlag(tlsFlag *string, baseUrl *url.URL) (*http.Client, bool, error) {
+	rootCAs, err := x509.SystemCertPool()
+	if err != nil {
+		log.Printf("initializing x509 system cert pool failed (%v), falling back to empty cert pool", err)
+	}
+	if rootCAs == nil {
+		rootCAs = x509.NewCertPool()
+	}
+
+	foundMatchingCertificate := false
+	// Append user specified certificate(s)
+	if *tlsFlag != "self-signed" {
+		usrCert := strings.Split(*tlsFlag, ",")[0]
+		certBytes, err := ioutil.ReadFile(usrCert)
+		if err != nil {
+			return nil, false, fmt.Errorf("reading certificate %s: %v", usrCert, err)
+		}
+		rootCAs.AppendCertsFromPEM(certBytes)
+	} else {
+		// Try to find a certificate in the local host config
+		hostConfig := config.HostnameSpecific(baseUrl.Host)
+		if err != nil {
```

Comment:
Please remove this obsolete error check, `config.HostnameSpecific` always succeeds